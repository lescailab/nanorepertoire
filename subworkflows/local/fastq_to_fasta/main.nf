// this subworkflow prepares the inputs from fastq files and does the translation, in the meanwhile performs a fastqc
// modules to include in this subworkflow

include {CUTADAPT       } from '../../../modules/local/cutadapt/main.nf'
include {FLASH          } from '../../../modules/nf-core/flash/main.nf'
include {RENAME         } from '../../../modules/local/rename/main.nf'
include {NANOTRANSLATE  } from '../../../modules/local/nanotranslate/main.nf'


///Users/bagordo/Desktop/all/all_bioinformatics/nf-core-nanorepertoire/data/*_{1,2}_dummy2.fastq
// main workflow
workflow FASTQ_TO_FASTA {

    // with take we define the input channels
    take:
    input          // channel: [[id], [reads_forward, reads_reverse]]
    adapterfile    // channel: "adapter.fasta"

    main:
    // define the channels that will be used to store the versions of the software

    ch_versions = Channel.empty()
    input_ch = input

    CUTADAPT(input_ch, adapterfile)
    ch_versions = ch_versions.mix(CUTADAPT.out.versions.first())

    FLASH(CUTADAPT.out.reads)
    ch_versions = ch_versions.mix(FLASH.out.versions.first())

    RENAME(FLASH.out.merged, single_end = true)
    ch_versions = ch_versions.mix(RENAME.out.versions.first())

    NANOTRANSLATE(RENAME.out.renamed)
    ch_versions = ch_versions.mix(NANOTRANSLATE.out.versions.first())

    // defining the output channels that the workflow will
    emit:
    // emitted channels
    trimmed_fastq = CUTADAPT.out.reads              // channel: [[id], [reads_forward, reads_reverse]]
    merged        = FLASH.out.merged                // channel: [[id], [ fastq_merged ]]
    renamed       = RENAME.out.renamed              // channel: [[id], [ fastq_renamed]]
    translated    = NANOTRANSLATE.out.fasta         // channel: [[id], [ fasta]]
    translog      = NANOTRANSLATE.out.log           // channel: [[id], [ fastalog]]
    versions      = ch_versions                     // channel: [ versions.yml ]
}

