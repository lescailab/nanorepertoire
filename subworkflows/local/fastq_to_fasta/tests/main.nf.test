// nf-core subworkflows test fastq_to_fasta
nextflow_workflow {

    name "Test Subworkflow FASTQ_TO_FASTA"
    script "../main.nf"
    config "./nextflow.config"
    workflow "FASTQ_TO_FASTA"

    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/fastq_to_fasta"
    tag "cutadapt"
    tag "flash"
    tag "rename"
    tag "nanotranslate"
    tag "fastqc"


    test("preprocessing - fastqc - fasta") {

        when {
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id: 'nftest', single_end:false ],
                        [
                            file("https://raw.githubusercontent.com/lescailab/testdata/master/nanobodies/test01_raw_R1.fastq.gz", checkIfExists: true),
                            file("https://raw.githubusercontent.com/lescailab/testdata/master/nanobodies/test01_raw_R2.fastq.gz", checkIfExists: true)
                        ]
                    ]
                )
                input[1] = Channel.of(
                    [
                        file("https://raw.githubusercontent.com/lescailab/testdata/master/nanobodies/adapters.fa", checkIfExists: true)
                    ]
                )
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }
}
