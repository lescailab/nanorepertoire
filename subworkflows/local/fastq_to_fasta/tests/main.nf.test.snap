{
    "preprocessing - fastqc - fasta": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "nftest",
                            "single_end": false
                        },
                        [
                            "nftest_1.trim.fastq.gz:md5,59230cd0ddccbcc2570da094a0d11979",
                            "nftest_2.trim.fastq.gz:md5,7a43f31adca405c0730727ca226b10fc"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "nftest",
                            "single_end": false
                        },
                        "nftest.extendedFrags.fastq.gz:md5,efd9425172c23697321b80db9f8688be"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "nftest",
                            "single_end": false
                        },
                        "nftest_renamed.fastq.gz:md5,de3ce43ade5486632270b146be13323d"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "nftest",
                            "single_end": false
                        },
                        "nftest_AA_translated.fasta:md5,0e80d34b88c3b8e449987070d99fb9ad"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "nftest",
                            "single_end": false
                        },
                        "nftest_AA_translated.log:md5,05aff95ed93fe9b6cffb5362c06a4041"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "nftest",
                            "single_end": false
                        },
                        [
                            "nftest_1_fastqc.zip:md5,3d3c069f3a829e4df1d93339b3b36f2c",
                            "nftest_2_fastqc.zip:md5,f5146c15f982acc49431b27ed536221f"
                        ]
                    ]
                ],
                "6": [
                    "versions.yml:md5,0befd310402fc319652349ea4e4df1d5",
                    "versions.yml:md5,58e62e03321dfed5515054e2b82059c2",
                    "versions.yml:md5,65a346cd7706fd5ba13de5b1ef30e7e4",
                    "versions.yml:md5,ba5fdbe5151a441ca33968edea203d88",
                    "versions.yml:md5,ca055137cc997222712d0123dce46f09"
                ],
                "fastqc": [
                    [
                        {
                            "id": "nftest",
                            "single_end": false
                        },
                        [
                            "nftest_1_fastqc.zip:md5,3d3c069f3a829e4df1d93339b3b36f2c",
                            "nftest_2_fastqc.zip:md5,f5146c15f982acc49431b27ed536221f"
                        ]
                    ]
                ],
                "merged": [
                    [
                        {
                            "id": "nftest",
                            "single_end": false
                        },
                        "nftest.extendedFrags.fastq.gz:md5,efd9425172c23697321b80db9f8688be"
                    ]
                ],
                "renamed": [
                    [
                        {
                            "id": "nftest",
                            "single_end": false
                        },
                        "nftest_renamed.fastq.gz:md5,de3ce43ade5486632270b146be13323d"
                    ]
                ],
                "translated": [
                    [
                        {
                            "id": "nftest",
                            "single_end": false
                        },
                        "nftest_AA_translated.fasta:md5,0e80d34b88c3b8e449987070d99fb9ad"
                    ]
                ],
                "translog": [
                    [
                        {
                            "id": "nftest",
                            "single_end": false
                        },
                        "nftest_AA_translated.log:md5,05aff95ed93fe9b6cffb5362c06a4041"
                    ]
                ],
                "trimmed_fastq": [
                    [
                        {
                            "id": "nftest",
                            "single_end": false
                        },
                        [
                            "nftest_1.trim.fastq.gz:md5,59230cd0ddccbcc2570da094a0d11979",
                            "nftest_2.trim.fastq.gz:md5,7a43f31adca405c0730727ca226b10fc"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0befd310402fc319652349ea4e4df1d5",
                    "versions.yml:md5,58e62e03321dfed5515054e2b82059c2",
                    "versions.yml:md5,65a346cd7706fd5ba13de5b1ef30e7e4",
                    "versions.yml:md5,ba5fdbe5151a441ca33968edea203d88",
                    "versions.yml:md5,ca055137cc997222712d0123dce46f09"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-15T17:01:43.485956"
    },
    "preprocessing - fastqc - both_end": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "nftest",
                            "single_end": false
                        },
                        [
                            "nftest_1.trim.fastq.gz:md5,59230cd0ddccbcc2570da094a0d11979",
                            "nftest_2.trim.fastq.gz:md5,7a43f31adca405c0730727ca226b10fc"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "nftest",
                            "single_end": false
                        },
                        "nftest.extendedFrags.fastq.gz:md5,7a9fedf788d42b46bfd36884adab7dfa"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "nftest",
                            "single_end": false
                        },
                        "nftest_renamed.fastq.gz:md5,e78f9ab4d574ef486a611d1ca52c86aa"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "nftest",
                            "single_end": false
                        },
                        "nftest_AA_translated.fasta:md5,803fd2dce066a8089ceb8a411f9777f6"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "nftest",
                            "single_end": false
                        },
                        "nftest_AA_translated.log:md5,05aff95ed93fe9b6cffb5362c06a4041"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "nftest",
                            "single_end": false
                        },
                        [
                            "nftest_1_fastqc.zip:md5,21cb8fc2594d65e2211ae743eae9429b",
                            "nftest_2_fastqc.zip:md5,61df84ebdbe4e3d5097bc7bf0dce26e5"
                        ]
                    ]
                ],
                "6": [
                    "versions.yml:md5,0befd310402fc319652349ea4e4df1d5",
                    "versions.yml:md5,58e62e03321dfed5515054e2b82059c2",
                    "versions.yml:md5,5d4e5b871f51af0a0a7d8419c6fe58b2",
                    "versions.yml:md5,ba5fdbe5151a441ca33968edea203d88",
                    "versions.yml:md5,ca055137cc997222712d0123dce46f09"
                ],
                "fastqc": [
                    [
                        {
                            "id": "nftest",
                            "single_end": false
                        },
                        [
                            "nftest_1_fastqc.zip:md5,21cb8fc2594d65e2211ae743eae9429b",
                            "nftest_2_fastqc.zip:md5,61df84ebdbe4e3d5097bc7bf0dce26e5"
                        ]
                    ]
                ],
                "merged": [
                    [
                        {
                            "id": "nftest",
                            "single_end": false
                        },
                        "nftest.extendedFrags.fastq.gz:md5,7a9fedf788d42b46bfd36884adab7dfa"
                    ]
                ],
                "renamed": [
                    [
                        {
                            "id": "nftest",
                            "single_end": false
                        },
                        "nftest_renamed.fastq.gz:md5,e78f9ab4d574ef486a611d1ca52c86aa"
                    ]
                ],
                "translated": [
                    [
                        {
                            "id": "nftest",
                            "single_end": false
                        },
                        "nftest_AA_translated.fasta:md5,803fd2dce066a8089ceb8a411f9777f6"
                    ]
                ],
                "translog": [
                    [
                        {
                            "id": "nftest",
                            "single_end": false
                        },
                        "nftest_AA_translated.log:md5,05aff95ed93fe9b6cffb5362c06a4041"
                    ]
                ],
                "trimmed_fastq": [
                    [
                        {
                            "id": "nftest",
                            "single_end": false
                        },
                        [
                            "nftest_1.trim.fastq.gz:md5,59230cd0ddccbcc2570da094a0d11979",
                            "nftest_2.trim.fastq.gz:md5,7a43f31adca405c0730727ca226b10fc"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0befd310402fc319652349ea4e4df1d5",
                    "versions.yml:md5,58e62e03321dfed5515054e2b82059c2",
                    "versions.yml:md5,5d4e5b871f51af0a0a7d8419c6fe58b2",
                    "versions.yml:md5,ba5fdbe5151a441ca33968edea203d88",
                    "versions.yml:md5,ca055137cc997222712d0123dce46f09"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-11T15:56:05.767921"
    }
}